File: examples/data/cdisc/01_basic_pc_ex/load.py
#!/usr/bin/env python3
"""
Basic CDISC Data Import - Python Example

Run: python load.py
"""

from neopkpd.data import load_cdisc


def main():
    print("CDISC PC/EX Data Import")
    print("=" * 50)

    # Load CDISC domains
    data = load_cdisc(
        pc="pc.csv",
        ex="ex.csv",
        dm="dm.csv"
    )

    print(f"\nData Summary:")
    print(f"  Subjects:     {len(data.subjects)}")
    print(f"  Observations: {len(data.observations)}")
    print(f"  Doses:        {len(data.doses)}")

    # Display subject information
    print("\nSubjects:")
    print("-" * 50)
    for subj in data.subjects:
        print(f"  {subj.id}:")
        print(f"    Age: {subj.covariates.get('age', 'N/A')}")
        print(f"    Sex: {subj.covariates.get('sex', 'N/A')}")
        print(f"    Observations: {len([o for o in data.observations if o.subject_id == subj.id])}")

    # Display observations for first subject
    print("\nObservations (Subject 1):")
    print("-" * 50)
    print("Time (h)  Concentration (ng/mL)")
    print("-" * 50)
    subj1_obs = [o for o in data.observations if o.subject_id == data.subjects[0].id]
    for obs in subj1_obs:
        print(f"  {obs.time:5.1f}     {obs.dv:8.2f}")

    # Display dosing
    print("\nDosing:")
    print("-" * 50)
    for dose in data.doses:
        print(f"  Subject {dose.subject_id}: {dose.amount} mg at t={dose.time}h ({dose.route})")

    # Convert to NeoPKPD format for analysis
    print("\n" + "=" * 50)
    print("NeoPKPD Format")
    print("=" * 50)

    neopkpd_data = data.to_neopkpd()
    print(f"\nReady for analysis with {len(neopkpd_data.subjects)} subjects")

    return data


if __name__ == "__main__":
    main()

File: examples/data/cdisc/01_basic_pc_ex/load.jl
# Basic CDISC Data Import - Julia Example
# Run: julia --project=packages/core load.jl

using NeoPKPD

println("CDISC PC/EX Data Import")
println("="^50)

# Load CDISC domains
data = load_cdisc(
    pc = "pc.csv",
    ex = "ex.csv",
    dm = "dm.csv"
)

println("\nData Summary:")
println("  Subjects:     $(length(data.subjects))")
println("  Observations: $(length(data.observations))")
println("  Doses:        $(length(data.doses))")

# Display subject information
println("\nSubjects:")
println("-"^50)
for subj in data.subjects
    println("  $(subj.id):")
    println("    Age: $(get(subj.covariates, "age", "N/A"))")
    println("    Sex: $(get(subj.covariates, "sex", "N/A"))")
    n_obs = count(o -> o.subject_id == subj.id, data.observations)
    println("    Observations: $n_obs")
end

# Display observations for first subject
println("\nObservations (Subject 1):")
println("-"^50)
println("Time (h)  Concentration (ng/mL)")
println("-"^50)
subj1_obs = filter(o -> o.subject_id == data.subjects[1].id, data.observations)
for obs in subj1_obs
    println("  $(lpad(string(round(obs.time, digits=1)), 5))     $(lpad(string(round(obs.dv, digits=2)), 8))")
end

# Display dosing
println("\nDosing:")
println("-"^50)
for dose in data.doses
    println("  Subject $(dose.subject_id): $(dose.amount) mg at t=$(dose.time)h ($(dose.route))")
end

# Convert to NeoPKPD format
println("\n" * "="^50)
println("NeoPKPD Format")
println("="^50)

neopkpd_data = to_neopkpd(data)
println("\nReady for analysis with $(length(neopkpd_data.subjects)) subjects")
